src/cc/bpf_module.cc

```diff
@@ -153,24 +153,90 @@ static void debug_printf(Module *mod, IRBuilder<> &B, const string &fmt, vector<
   B.CreateCall(fprintf_fn, args);
 }
 
+static void finish_sscanf(IRBuilder<> &B, vector<Value *> *args, string *fmt,
```

Comment:
This is a genius hack! I just learned there exists %n for sscanf today! But the implementation is indeed very convoluted. Do you want to give a little more information like below about how reader and writer are generated.
The comments to illustrate how below (or similar) is generated will be good:
typedef struct {
  char stack1[STACK_MAX_LEN];
} Aux;

typedef struct {
  uint32_t a;
  Aux stack1[STACK_MAX_LEN];
  uint32_t b;
} Event;
Event type is used as key or value type.
 
fmt = "{ %i [ { \"%n"; sscanf(...)

/* the following two lines repeated a few times */
fmt = "%[^\"]%n"; sscanf(...)
fmt = "\" } { \"%n"; sscanf(...)

fmt = ""\" } ] %i }%n"; sscanf(...)

hopefully I did not make any mistake.